terraform/modules/jenkins/jenkins/ec2.tf

```diff
@@ -4,6 +4,7 @@ provider "aws" {
 
 resource "aws_instance" "jenkins" {
   ami                    = "${var.ami_id}"
+  disable_api_termination= true  # prevent this instance from being destroyed from the console
```

Comment:
Does this also prevent us destroying the instance with Terraform? Would we have to change & apply this attribute first if we needed to destroy for some reason?

Reply:
I believe if for some reason we wanted to destroy the instance we would just remove it from the terraform file, and then terraform would see that it was terminated.

Alternatively we could terminate it from the AWS console by first removing the termination protection from the AWS console (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/terminating-instances.html#Using_ChangingDisableAPITermination). This information is in the terraform manual for the aws_ec2_instance resource.

Reply:
Hmm, ok. Let's make sure we note this down in the manual.